tools/java/org/unicode/cldr/util/LogicalGrouping.java

```diff
@@ -199,6 +199,13 @@ public static boolean isOptional(CLDRFile cldrFile, String path) {
         if (pluralType.equals("0") || pluralType.equals("1")) return true;
         if (!pluralType.equals("zero") && !pluralType.equals("one")) return false;
 
+        if (pluralType.equals("many")) { // special case for french
+            String localeId = cldrFile.getLocaleID();
+            if (localeId.startsWith("fr")
+                && (localeId.length() == 2 || localeId.charAt(2) == '_'))
```

Comment:
using CLDRLocale this is `if(localeId.childOf(fr))`… where fr is a singleton. Should revisit someday.